src/compiler/checker.ts

```diff
@@ -1100,7 +1098,7 @@ namespace ts {
         }
 
         function getDeclarationOfAliasSymbol(symbol: Symbol): Declaration | undefined {
-            return forEach(symbol.declarations, d => isAliasSymbolDeclaration(d) ? d : undefined);
+            return find<Declaration>(symbol.declarations, isAliasSymbolDeclaration);

```

Comment:
Probably yes -- I would think that `symbol.declarations` would be the one providing the type for inference. It's worth investigating even if it ends up being by design.